src/routers/alpha-router/quoters/mixed-quoter.ts

```diff
@@ -186,14 +218,25 @@ export class MixedQuoter extends BaseQuoter<
       maxSwapsPerPath
     );
 
+    // special case mixed route w/ native currency routing
+    // there's a possibility upstream is requesting the native currency,
+    // but we might want to wrap into the wrapped token because maybe only v2/v3 have the routes.
+    // in that case, we have to remain the non-native currency routing as is today for mixed routing.
+    const v2V3OnlyRoutes = computeAllMixedRoutes(
+      currencyIn.wrapped,
+      currencyOut.wrapped,
+      v2V3OnlyPools,
+      maxSwapsPerPath
+    );
+
     metric.putMetric(
       'MixedGetRoutesLoad',
       Date.now() - beforeGetRoutes,
       MetricLoggerUnit.Milliseconds
     );
 
     return {
-      routes: routes,
+      routes: routes.concat(v2V3OnlyRoutes),
```

Comment:
meanwhile dup is not a problem, because best-swap-route will select a single best route, it's better if we do dedup here. I will think about how to do it here.